cluster-provision/images/vm-image-builder/create-containerdisk.sh

```diff
@@ -6,90 +6,105 @@ if [ "$#" -ne 1 ]; then
+
 export IMAGE_NAME=$1
 
 if [ -f "${SCRIPT_PATH}/${IMAGE_NAME}/create-image.sh" ]; then
     export TAG="devel"
     readonly VM_IMAGE="provisioned-image.qcow2"
     readonly build_directory="${SCRIPT_PATH}/${IMAGE_NAME}/build"
-    
+
     trap 'cleanup' EXIT SIGINT
-    
+
     mkdir "${build_directory}"
 
     pushd "${SCRIPT_PATH}/${IMAGE_NAME}"
       cleanup
       echo "Creating the image"
-      ./create-image.sh "${build_directory}/${VM_IMAGE}" 
+      ./create-image.sh "${build_directory}/${VM_IMAGE}"
 
       echo "Creating the containerdisk ..."
       docker build . -t ${IMAGE_NAME}:${TAG} -f - <<END
 FROM scratch
 ADD --chown=107:107 build/${VM_IMAGE} /disk/
 END
     popd
-
-else 
-    export TAG=devel
+else
     export OS_VARIANT="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/os-variant)"
     export CLOUD_CONFIG_PATH="${SCRIPT_PATH}/${IMAGE_NAME}/cloud-config"
-    export VM_IMAGE_URL="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url)"
-
-    readonly VM_IMAGE="source-image.qcow2"
-    readonly build_directory="${IMAGE_NAME}_build"
-    readonly new_vm_image_name="provisioned-image.qcow2"
 
+    if [[ ${ARCH} = "aarch64" ]]; then
+        export VM_IMAGE_URL="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url-aarch64)"
+        export TAG=devel-aarch64
+        readonly VM_IMAGE="source-image-aarch64.qcow2"
+        readonly build_directory="${IMAGE_NAME}_build_aarch64"
+        readonly new_vm_image_name="provisioned-image-aarch64.qcow2"
+    else
+        export VM_IMAGE_URL="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url)"
+        export TAG=devel
+        readonly VM_IMAGE="source-image.qcow2"
+        readonly build_directory="${IMAGE_NAME}_build"
+        readonly new_vm_image_name="provisioned-image.qcow2"
+    fi
     trap 'cleanup' EXIT SIGINT
 
     pushd "${SCRIPT_PATH}"
       cleanup
       echo "Downloading the base image ..."
 
-       if ! [ -e "${VM_IMAGE}" ]; then
-        # Download base VM image
-        curl -L "${VM_IMAGE_URL}" -o "${VM_IMAGE}"
+      if ! [ -e "${VM_IMAGE}" ]; then
+          # Download base VM image
+          curl -L "${VM_IMAGE_URL}" -o "${VM_IMAGE}"
       fi
 
       mkdir "${build_directory}"
 
       echo "Running the image customization ..."
-      customize_image "${VM_IMAGE}" "${OS_VARIANT}" "${build_directory}/${new_vm_image_name}" "${CLOUD_CONFIG_PATH}"
+      ARCH=${ARCH} customize_image "${VM_IMAGE}" "${OS_VARIANT}" "${build_directory}/${new_vm_image_name}" "${CLOUD_CONFIG_PATH}"
 
       echo "Creating the containerdisk ..."
-      docker build . -t ${IMAGE_NAME}:${TAG} -f - <<END
+      if [ ${ARCH} = "aarch64" ]; then
+          DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --platform "linux/arm64" . -t ${IMAGE_NAME}:${TAG} -f - <<END
 FROM scratch
 ADD --chown=107:107 ${build_directory}/${new_vm_image_name} /disk/
 END
+      else
+          docker build . -t ${IMAGE_NAME}:${TAG} -f - <<END
+FROM scratch
+ADD --chown=107:107 ${build_directory}/${new_vm_image_name} /disk/
+END
```

Comment:
Please consider setting a variable before the `IF` that will change docker command according to $ARCH
and extract this to function, something like:
```
function build_container() {
    local -r build_directory=$1
    local -r new_vm_image_name=$2
    local -r arch=$3
    local -r image_name=$4
    local -r tag=$5

    if [ $arch = "aarch64" ]; then
        local image_tag="$tag-$arch"
        docker_build="DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --platform "linux/arm64" . -t ${image_name}:${image_tag}"
    else 
        docker_build=docker build . -t ${image_name}:${tag}"
    fi
    $docker_build -f - <<END
FROM scratch
ADD --chown=107:107 ${build_directory}/${new_vm_image_name} /disk/
END
}
...
# usage:
build_container "$build_directory" "$new_vm_image_name" "$ARCH" "$IMAGE_NAME" "$TAG"
```